services/credits-service/src/middleware/auth.py

```
@@ -0,0 +1,96 @@
+from typing import Callable
+
+from fastapi import Request, HTTPException, status
+from starlette.middleware.base import BaseHTTPMiddleware
+
+logger = logging.getLogger(__name__)
+
+
+class APIKeyAuthMiddleware(BaseHTTPMiddleware):
+    """Middleware to validate API keys for protected endpoints"""
+
+    def __init__(self, app, exempt_paths: list[str] = None):
```

Ruff (0.14.7)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)